### DISCUSSION TITLE
how to use onError callback only when fetchNextPage has failed?

### DISCUSSION THREAD

**Discussion by emewjin** [TanStack/query#5403] [Q&A]
Hi! 
When using `useInfiniteQuery`, I want the `onError` callback to be triggered only when `fetchNextPage` fails, except for the case where the initial page fetch fails.
Is there a recommended way or best practice for this?

**Comment by TkDodo** [comment]
onError is deprecated and will be removed in the next major version:  

- https://github.com/TanStack/query/discussions/5279

**Reply by emewjin** [reply]
wow thanks
